destroyScreen, screenAdded removed from Qt 5.12

It appears that screenAdded and destroyAdded are no longer a thing

```
/tmp/qtmir/src/platforms/mirserver/mirserverintegration.cpp: In lambda function:
/tmp/qtmir/src/platforms/mirserver/mirserverintegration.cpp:160:15: error: ‘class MirServerIntegration’ has no member named ‘destroyScreen’
  160 |         this->destroyScreen(screen);
      |               ^~~~~~~~~~~~~
/tmp/qtmir/src/platforms/mirserver/mirserverintegration.cpp: In member function ‘virtual void MirServerIntegration::initialize()’:
/tmp/qtmir/src/platforms/mirserver/mirserverintegration.cpp:165:9: error: ‘screenAdded’ was not declared in this scope
  165 |         screenAdded(screen);
      |         ^~~~~~~~~~~
```

--- a/src/platforms/mirserver/mirserverintegration.cpp
+++ b/src/platforms/mirserver/mirserverintegration.cpp
@@ -151,18 +151,18 @@ void MirServerIntegration::initialize()
         qFatal("ScreensModel not initialized");
     }
     QObject::connect(screens.data(), &ScreensModel::screenAdded,
-            [this](Screen *screen) { this->screenAdded(screen); });
+            [this](Screen *screen) { QWindowSystemInterface::handleScreenAdded(screen); });
     QObject::connect(screens.data(), &ScreensModel::screenRemoved,
             [this](Screen *screen) {
 #if QT_VERSION < QT_VERSION_CHECK(5, 5, 0)
         delete screen;
 #else
-        this->destroyScreen(screen);
+        QWindowSystemInterface::handleScreenRemoved(screen);
 #endif
     });
 
     Q_FOREACH(auto screen, screens->screens()) {
-        screenAdded(screen);
+        QWindowSystemInterface::handleScreenAdded(screen);
     }
 
     m_nativeInterface = new NativeInterface(m_mirServer.data());

--- a/src/platforms/mirserver/mirserverintegration.cpp
+++ b/src/platforms/mirserver/mirserverintegration.cpp
@@ -143,7 +143,7 @@ QAbstractEventDispatcher *MirServerIntegration::createEventDispatcher() const
 
 void MirServerIntegration::handleScreenAdded(QPlatformScreen *screen) const
 {
-    # TODO: remove this after we no longer support Qt 5.13
+    // TODO: remove this after we no longer support Qt 5.13
     #if QT_VERSION < QT_VERSION_CHECK(5, 13, 0)
             this->screenAdded(screen);
     #else
@@ -153,7 +153,7 @@ void MirServerIntegration::handleScreenAdded(QPlatformScreen *screen) const
 
 void MirServerIntegration::handleScreenRemoved(QPlatformScreen *screen) const
 {
-    # TODO: remove this after we no longer support Qt 5.13
+    // TODO: remove this after we no longer support Qt 5.13
     #if QT_VERSION < QT_VERSION_CHECK(5, 13, 0)
             this->destroyScreen(screen);
     #else

--- a/src/platforms/mirserver/mirserverintegration.cpp
+++ b/src/platforms/mirserver/mirserverintegration.cpp
@@ -141,6 +141,26 @@ QAbstractEventDispatcher *MirServerIntegration::createEventDispatcher() const
     return createUnixEventDispatcher();
 }
 
+void MirServerIntegration::handleScreenAdded(QPlatformScreen *screen) const
+{
+    # TODO: remove this after we no longer support Qt 5.13
+    #if QT_VERSION < QT_VERSION_CHECK(5, 13, 0)
+            this->screenAdded(screen);
+    #else
+            QWindowSystemInterface::handleScreenAdded(screen);
+    #endif
+}
+
+void MirServerIntegration::handleScreenRemoved(QPlatformScreen *screen) const
+{
+    # TODO: remove this after we no longer support Qt 5.13
+    #if QT_VERSION < QT_VERSION_CHECK(5, 13, 0)
+            this->destroyScreen(screen);
+    #else
+            QWindowSystemInterface::handleScreenRemoved(screen);
+    #endif
+}
+
 void MirServerIntegration::initialize()
 {
     // Creates instance of and start the Mir server in a separate thread
@@ -151,18 +171,18 @@ void MirServerIntegration::initialize()
         qFatal("ScreensModel not initialized");
     }
     QObject::connect(screens.data(), &ScreensModel::screenAdded,
-            [this](Screen *screen) { QWindowSystemInterface::handleScreenAdded(screen); });
+            [this](Screen *screen) { handleScreenAdded(screen); });
     QObject::connect(screens.data(), &ScreensModel::screenRemoved,
             [this](Screen *screen) {
 #if QT_VERSION < QT_VERSION_CHECK(5, 5, 0)
         delete screen;
 #else
-        QWindowSystemInterface::handleScreenRemoved(screen);
+        handleScreenRemoved(screen);
 #endif
     });
 
     Q_FOREACH(auto screen, screens->screens()) {
-        QWindowSystemInterface::handleScreenAdded(screen);
+        handleScreenAdded(screen);
     }
 
     m_nativeInterface = new NativeInterface(m_mirServer.data());

--- a/src/platforms/mirserver/mirserverintegration.cpp
+++ b/src/platforms/mirserver/mirserverintegration.cpp
@@ -141,7 +141,7 @@ QAbstractEventDispatcher *MirServerIntegration::createEventDispatcher() const
     return createUnixEventDispatcher();
 }
 
-void MirServerIntegration::handleScreenAdded(QPlatformScreen *screen) const
+void MirServerIntegration::handleScreenAdded(QPlatformScreen *screen)
 {
     // TODO: remove this after we no longer support Qt 5.13
     #if QT_VERSION < QT_VERSION_CHECK(5, 13, 0)
@@ -151,7 +151,7 @@ void MirServerIntegration::handleScreenAdded(QPlatformScreen *screen) const
     #endif
 }
 
-void MirServerIntegration::handleScreenRemoved(QPlatformScreen *screen) const
+void MirServerIntegration::handleScreenRemoved(QPlatformScreen *screen)
 {
     // TODO: remove this after we no longer support Qt 5.13
     #if QT_VERSION < QT_VERSION_CHECK(5, 13, 0)

--- a/src/platforms/mirserver/mirserverintegration.h
+++ b/src/platforms/mirserver/mirserverintegration.h
@@ -37,8 +37,8 @@ public:
     QPlatformOpenGLContext *createPlatformOpenGLContext(QOpenGLContext *context) const override;
 
     QAbstractEventDispatcher *createEventDispatcher() const override;
-    void handleScreenAdded(QPlatformScreen *screen) const;
-    void handleScreenRemoved(QPlatformScreen *screen) const;
+    void handleScreenAdded(QPlatformScreen *screen);
+    void handleScreenRemoved(QPlatformScreen *screen);
     void initialize() override;
 
     QPlatformClipboard *clipboard() const override;

--- a/src/platforms/mirserver/mirserverintegration.h
+++ b/src/platforms/mirserver/mirserverintegration.h
@@ -37,6 +37,8 @@ public:
     QPlatformOpenGLContext *createPlatformOpenGLContext(QOpenGLContext *context) const override;
 
     QAbstractEventDispatcher *createEventDispatcher() const override;
+    void handleScreenAdded(QPlatformScreen *screen) const;
+    void handleScreenRemoved(QPlatformScreen *screen) const;
     void initialize() override;
 
     QPlatformClipboard *clipboard() const override;

